File: problems/0001_two-sum/Solution.php
<?php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $m = [];
        $l = count($nums);
        for ($i = 0; $i < $l; ++$i) {
            $num = $nums[$i];
            $r = $target - $num;
            if (array_key_exists($r, $m))
                return [$m[$r], $i];
            $m[$num] = $i;
        }
        throw new Exception('No match value for $nums, $target!');
    }
}

File: problems/0001_two-sum/Solution.py
#!/usr/bin/python3

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {};
        for i, num in enumerate(nums):
            r = target - num;
            if r in m:
                return [m[r], i];
            m[num] = i;
        raise Exception(f"No match value for {nums}, {target}!");
